### PR TITLE
add initial support environment variables

### PR DESCRIPTION
**Pull request by jbovet**
## This PR

This PR adds a basic implementation of the Environment Variables feature provider

## Supported Types

The provider supports the following types:

- Int
- float
- String
- bool

> Please note that *Struct* type is not currently supported yet.

doc: https://github.com/jbovet/rust-sdk-contrib/blob/main/crates/env-var/README.md
changelog: https://github.com/jbovet/rust-sdk-contrib/blob/main/crates/env-var/CHANGELOG.md

### Related Issues

related [#339](https://github.com/open-feature/java-sdk/issues/339)

### How to test

cucumber features supported

### CODE REVIEW

File: crates/env-var/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+let mut api = OpenFeature::singleton_mut().await;
+let provider  = EnvVarProvider::default();
+api.set_provider(provider).await;
+let client = api.create_named_client("env-var-client");
+
+let mut message =  "Hello rustaceans!";
+let is_feature_enabled = client.get_bool_value("env-flag-key", &EvaluationContext::default(), None).await.unwrap_or(false);
+
+if is_feature_enabled {
+    message = "Hello rustaceans from feature flag!";
```
Comment (beeme1mr) [inline crates/env-var/README.md:42]: Nice 👍

**Review by beeme1mr — APPROVED** [review]
I'm not a rustaceans but this looks good to me. I also noticed that we're not running tests, so I created a few CI-related issues to implement it.

I'll approve but wait for feedback from our resident Rust expert @sheepduke.

**Review by sheepduke — APPROVED** [review]

**Comment by sheepduke** [comment]
My apologies for the delayed response! :-)

File: crates/env-var/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    ///         .resolve_string_value(flag_key, &EvaluationContext::default())
+    ///         .await;
+    ///     assert!(res.is_ok());
+    ///     assert_eq!(res.unwrap().value, value);
+    /// }
+    /// ```
+    async fn resolve_bool_value(
+        &self,
+        flag_key: &str,
+        _evaluation_context: &EvaluationContext,
```
[the remark this answers is not in the document]
Comment (jbovet) [reply crates/env-var/src/lib.rs]: Yes, you're right, renamed evaluation context variable, thanks

File: crates/env-var/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+///     let res = provider.resolve_string_value(flag_key, &EvaluationContext::default()).await;
+///     assert!(res.is_err());
+///     assert_eq!(res.unwrap_err().code, EvaluationErrorCode::FlagNotFound);
+/// }
+/// ```
+fn error<T>(evaluation_error_code: EvaluationErrorCode) -> EvaluationResult<T> {
+    Err(EvaluationError::builder()
+        .message("Error evaluating environment variable")
+        .code(evaluation_error_code)
+        .build())
```
[the remark this answers is not in the document]
Comment (jbovet) [reply crates/env-var/src/lib.rs:214]: I felt the same when I tried to build the error handler, in fact, the builder pattern is certainly much more verbose, but it provides some advantages that can make it worth using, I will take a look at the SDK, maybe I can help with this.

[the remark this answers is not in the document]
Comment (jbovet) [reply crates/env-var/src/lib.rs:214]: hi @sheepduke, coudl we implement simpler error creation for EvaluationError like this? https://github.com/open-feature/rust-sdk/commit/66d6eb62e675ab80ace1c4d9ef89552787440a97 and you can use in this way https://github.com/open-feature/rust-sdk/commit/66d6eb62e675ab80ace1c4d9ef89552787440a97#diff-b6cb772ec8f17d2f5f9d9a3ac09a9fb9bf4d68120860aa0c744c220273523583R81

[the remark this answers is not in the document]
Comment (sheepduke) [reply crates/env-var/src/lib.rs:214]: > hi @sheepduke, could we implement simpler error creation for EvaluationError like this? [open-feature/rust-sdk@66d6eb6](https://github.com/open-feature/rust-sdk/commit/66d6eb62e675ab80ace1c4d9ef89552787440a97) and you can use in this way [open-feature/rust-sdk@66d6eb6#diff-b6cb772ec8f17d2f5f9d9a3ac09a9fb9bf4d68120860aa0c744c220273523583R81](https://github.com/open-feature/rust-sdk/commit/66d6eb62e675ab80ace1c4d9ef89552787440a97#diff-b6cb772ec8f17d2f5f9d9a3ac09a9fb9bf4d68120860aa0c744c220273523583R81) and [open-feature/rust-sdk@66d6eb6#diff-dd115732bc0121b31e06a3f035822d39790e57aa863d093d3c5ce8a7d10cbc6cR184](https://github.com/open-feature/rust-sdk/commit/66d6eb62e675ab80ace1c4d9ef89552787440a97#diff-dd115732bc0121b31e06a3f035822d39790e57aa863d093d3c5ce8a7d10cbc6cR184)

I thought of this solution, but...I was afraid that the EvaluationError specification would change in the future. If that happens, we have to modify the current API due to lack of function overloading.

[the remark this answers is not in the document]
Comment (jbovet) [reply crates/env-var/src/lib.rs:214]: I agree, builder is more flexible!

**Comment by jbovet** [comment]
@beeme1mr @sheepduke could we merge this PR?

**Comment by beeme1mr** [comment]
Hey @jbovet, sorry for the delay. I've merged this PR. Thanks!

**Comment by jbovet** [comment]
> Hey @jbovet, sorry for the delay. I've merged this PR. Thanks!

Happy to help @beeme1mr

**Pull request merged**